aws/resource_aws_qldb_ledger_test.go

```diff
@@ -0,0 +1,211 @@
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSQLDBLedgerDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSQLDBLedgerConfig(rInt),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSQLDBLedgerExists(resourceName, &qldbCluster),
+					resource.TestMatchResourceAttr(resourceName, "arn", regexp.MustCompile(`^arn:[^:]+:qldb:[^:]+:\d{12}:ledger/.+`)),
+					resource.TestMatchResourceAttr(resourceName, "name", regexp.MustCompile("test-ledger-[0-9]+")),
+					resource.TestCheckResourceAttr(resourceName, "permissions_mode", "ALLOW_ALL"),
+					resource.TestCheckResourceAttr(resourceName, "deletion_protection", "false"),
+					resource.TestCheckResourceAttr(resourceName, "tags.%", "0"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+				ImportStateVerifyIgnore: []string{
+					"apply_immediately",
+				},
+			},
+		},
+	})
+}
+
+func testAccCheckAWSQLDBLedgerDestroy(s *terraform.State) error {
+	err := testAccCheckAWSClusterDestroyWithProvider(s, testAccProvider)
+	if err != nil {
+		qldberr := err.(awserr.Error)
+		if qldberr.Code() == "ResourceInUseException" {
+			log.Println("Ledger is in creating state. Retrying in 10 seconds...")
+			time.Sleep(time.Second * 10)
+			return testAccCheckAWSQLDBLedgerDestroy(s)
+		}
+	}
+
+	return err
+}
+
+func testAccCheckAWSQLDBLedgerExists(n string, v *qldb.DescribeLedgerOutput) resource.TestCheckFunc {
+	return testAccCheckAWSQLDBLedgerExistsWithProvider(n, v, func() *schema.Provider { return testAccProvider })
+}
+
+func testAccCheckAWSQLDBLedgerExistsWithProvider(n string, v *qldb.DescribeLedgerOutput, providerF func() *schema.Provider) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No QLDB Ledger ID is set")
+		}
+
+		provider := providerF()
+		conn := provider.Meta().(*AWSClient).qldbconn
+		resp, err := conn.DescribeLedger(&qldb.DescribeLedgerInput{
+			Name: aws.String(rs.Primary.ID),
+		})
+
+		if err != nil {
+			return err
+		}
+
+		if *resp.Name == rs.Primary.ID {
+			*v = *resp
+			return nil
+		}
+
+		return fmt.Errorf("QLDB Ledger (%s) not found", rs.Primary.ID)
+	}
+}
+
+func testAccAWSQLDBLedgerConfig(n int) string {
+	return fmt.Sprintf(`
+resource "aws_qldb_ledger" "test" {
+  name                            = "test-ledger-%d"
+  deletion_protection             = false
+}
+`, n)
+}
+
+func TestAccAWSQLDBLedger_Tags(t *testing.T) {
+	var cluster1, cluster2, cluster3 qldb.DescribeLedgerOutput
+	rName := acctest.RandomWithPrefix("tf-acc-test")
+	resourceName := "aws_qldb_ledger.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t); testAccPreCheckAWSEks(t) },
```

Comment:
Copy-paste of wrong service PreCheck:

```suggestion
		PreCheck:     func() { testAccPreCheck(t) },
```